### PR TITLE
Add fuzz tests

### PR DESCRIPTION
**Pull request by qiwzhang**
Signed-off-by: Wayne Zhang <qiwzhang@google.com>

### CODE REVIEW

File: test/fuzz/fuzz_test.cc [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+#include <string>
+
+namespace google {
+namespace jwt_verify {
+namespace {
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  // split the data int "jwt . jwks".
+  std::vector<std::string> v = absl::StrSplit(std::string((const char*)data, size), ".");
+  // jwt has 2 dot, at least we should have 3 dots with 4 sections.
```
Comment (nareddyt) [inline test/fuzz/fuzz_test.cc:33]: Too much validation? Fuzzer should pass these invalid inputs (JWTs that don't have 3 dots) to the JWT verify lib. It is realistic input the lib might receive

File: test/fuzz/BUILD [hunk 2]
```diff
@@ -0,0 +1,19 @@
+licenses(["notice"])
+
+package(default_visibility = ["//visibility:public"])
+
+exports_files(["LICENSE"])
+
+load("@rules_fuzzing//fuzzing:cc_defs.bzl", "cc_fuzz_test")
+
+cc_fuzz_test(
+    name = "fuzz_test",
```
[the remark this answers is not in the document]
Comment (qiwzhang) [reply test/fuzz/BUILD]: Done

File: test/fuzz/fuzz_test.cc [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+#include <cstdint>
+#include <cstddef>
+#include <string>
+
+namespace google {
+namespace jwt_verify {
+namespace {
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
```
[the remark this answers is not in the document]
Comment (qiwzhang) [reply test/fuzz/fuzz_test.cc]: Done

File: test/fuzz/fuzz_test.cc (hunk 1 above)
Comment (qiwzhang) [reply test/fuzz/fuzz_test.cc]: Updated to use libprotobuf-mutator

**Comment by qiwzhang** [comment]
Two issues remaining
* there are a lot of text_format errors when running the fuzz tests.
* not sure if the corpus files format is good.  since jwks is a JSON with double quote,  I used single quota. not sure if it works

File: test/fuzz/corpus/jwt_verify_lib_fuzz_test/jwks_ec.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+jwt: "base64ed+header.base64+claims.base64+signature"
+jwks: '{"keys":[{"kty":"EC","crv":"P-256","alg":"ES256","kid":"abc","x":"EB54wykhS7YJFD6RYJNnwbWEz3cI7CF5bCDTXlrwI5k","y":"92bCBTvMFQ8lKbS2MbgjT3YfmYo6HnPEE2tsAqWUJw8"},{"kty":"EC","crv":"P-256","alg":"ES256","kid":"xyz","x":"EB54wykhS7YJFD6RYJNnwbWEz3cI7CF5bCDTXlrwI5k","y":"92bCBTvMFQ8lKbS2MbgjT3YfmYo6HnPEE2tsAqWUJw8"},{"kty":"EC","crv":"P-384","alg":"ES384","kid":"es384","x":"yY8DWcyWlrr93FTrscI5Ydz2NC7emfoKYHJLX2dr3cSgfw0GuxAkuQ5nBMJmVV5g","y":"An5wVxEfksDOa_zvSHHGkeYJUfl8y11wYkOlFjBt9pOCw5-RlfZgPOa3pbmUquxZ"},{"kty":"EC","crv":"P-521","alg":"ES512","kid":"es512","x":"Abijiex7rz7t-_Zj_E6Oo0OXe9C_-MCSD-OWio15ATQGjH9WpbWjN62ZqrrU_nwJiqqwx6ZsYKhUc_J3PRaMbdVC","y":"FxaljCIuoVEA7PJIaDPJ5ePXtZ0hkinT1B_bQ91mShCiR_43Whsn1P7Gz30WEnLuJs1SGVz1oT4lIRUYni2OfIk"}]}'
```
Comment (nareddyt) [inline test/fuzz/corpus/jwt_verify_lib_fuzz_test/jwks_ec.txt:2]: Based on https://developers.google.com/protocol-buffers/docs/text-format-spec#string, I don't think single quote in textformat allows you to use unescaped `"` or `\n` characters.

Suggest you escape the string (you can use "moma/cyberchef").

**Review by nareddyt — COMMENTED** [review]
Try fixing the escaping comment. Let's not worry too much about libprotobuf ERRORs. It should still eventually be triggering some valid inputs. Once we have OSS fuzz integration we can verify code coverage of fuzzer, and go back to fix it if there is none.

File: test/fuzz/corpus/jwt_verify_lib_fuzz_test/jwks_ec.txt (hunk 4 above)
Comment (qiwzhang) [reply test/fuzz/corpus/jwt_verify_lib_fuzz_test/jwks_ec.txt:2]: I just added a test to test all corpus file formats.

**Review by nareddyt — APPROVED** [review]

**Pull request merged**